src/arkode/arkode_sprkstep.c

```diff
@@ -561,19 +561,23 @@ int sprkStep_TakeStep(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
     SUNLogExtraDebugVec(ARK_LOGGER, "stage", prev_stage, "z2_%i(:) =", is);
 
     /* evaluate p' with the previous velocity */
-    N_VConst(ZERO, step_mem->sdata); /* either have to do this or ask user to
-                                        set other outputs to zero */
-    retval = sprkStep_f1(step_mem, ark_mem->tn + chati * ark_mem->h, prev_stage,
-                         step_mem->sdata, ark_mem->user_data);
+    if (ahati != 0.0)
```

Comment:
```suggestion
    if (SUNRabs(ahati) <= TINY)
```
If we want parity with
https://github.com/LLNL/sundials/blob/c8dabcea90ca8bf195474da120e4f3dd39aa711f/src/arkode/arkode_arkstep.c#L1901

Reply:
If we go with this recommendation, then a similar change is needed in line 593:
```
if (ai != 0.0)
```
(and similarly for lines 695 and 728)